tracer/src/Datadog.Trace/ClrProfiler/AutomaticTracer.cs

```diff
@@ -61,19 +62,14 @@ void IDistributedTracer.SetSpanContext(IReadOnlyDictionary<string, string> value
             }
         }
 
-        void IDistributedTracer.LockSamplingPriority()
+        SamplingPriority? IDistributedTracer.GetSamplingPriority()
```

Comment:
Can we go further than "remove locking" and also remove all synchronization between automatic and manual tracers (and let each one make their own sampling decision)?

Reply:
(can be in separate PR)

Reply:
Why would you want to do that? In the common case, when setting the sampling priority using the manual instrumentation API, you would expect it to apply to the whole trace (thus requiring synchronization between the tracers)

Reply:
> why you would want to do that?

To make things simpler by removing a lot of the synchronization code. We will send separate traces for automatic and manual spans, and it's ok for them to have different sampling decisions. 

It will be the equivalent of having two services making different sampling decisions. For example, if `ServiceA` has automatic instrumentation and chooses `AUTO_REJECT`, and it calls `ServiceB` and the user sets `USER_KEEP` manually there. The backend will connect the distributed trace and figure out a final sampling decision for the entire trace.

Reply:
> and it's ok for them to have different sampling decisions

But, in the common case, where they have the same configuration, they will make the same sampling decision anyway.

Reply:
Never mind. Thanks for pointing out that the Agent would in most cases drop the trace with `AUTO_REJECT` in the example above.

> when setting the sampling priority using the manual instrumentation API, I would expect it to apply to the whole trace

I agree with this. From the user's point of view, it's all one trace, even if internally we split it into multiple traces. Resolving this comment.